src/promptflow-evals/promptflow/evals/evaluate/_eval_run.py

```diff
@@ -285,42 +323,74 @@ def log_artifact(self, artifact_folder: str) -> None:
         if not os.listdir(artifact_folder):
             LOGGER.error("The path to the artifact is empty.")
             return
+        if not os.path.isfile(os.path.join(artifact_folder, EvalRun.EVALUATION_ARTIFACT)):
+            LOGGER.error("The run results file was not found, skipping artifacts upload.")
+            return
         # First we will list the files and the appropriate remote paths for them.
-        upload_path = os.path.basename(os.path.normpath(artifact_folder))
-        remote_paths = {"paths": []}
+        root_upload_path = posixpath.join("promptflow", 'PromptFlowArtifacts', self.info.run_name)
+        remote_paths = {'paths': []}
         local_paths = []
-
+        # Go over the artifact folder and upload all artifacts.
         for (root, _, filenames) in os.walk(artifact_folder):
+            upload_path = root_upload_path
             if root != artifact_folder:
                 rel_path = os.path.relpath(root, artifact_folder)
-                if rel_path != ".":
-                    upload_path = posixpath.join(upload_path, rel_path)
+                if rel_path != '.':
+                    upload_path = posixpath.join(root_upload_path, rel_path)
             for f in filenames:
                 remote_file_path = posixpath.join(upload_path, f)
                 remote_paths["paths"].append({"path": remote_file_path})
                 local_file_path = os.path.join(root, f)
                 local_paths.append(local_file_path)
-        # Now we need to reserve the space for files in the artifact store.
-        headers = {
-            "Content-Type": "application/json",
-            "Accept": "application/json",
-            "Content-Length": str(len(json.dumps(remote_paths))),
-            "x-ms-client-request-id": str(uuid.uuid1()),
-        }
+
+        # We will write the artifacts to the workspaceblobstore
+        datastore = self._ml_client.datastores.get_default(include_secrets=True)
+        account_url = f"{datastore.account_name}.blob.{datastore.endpoint}"
+        svc_client = BlobServiceClient(
+            account_url=account_url, credential=self._get_datastore_credential(datastore))
+        for local, remote in zip(local_paths, remote_paths['paths']):
+            blob_client = svc_client.get_blob_client(
+                container=datastore.container_name,
+                blob=remote['path'])
+            with open(local, 'rb') as fp:
+                blob_client.upload_blob(fp, overwrite=True)

```

Comment:
All this logic exist in pf-azure. Would it be beneficial to reuse it ?

Reply:
PF azure logic is dependent on RunOperations class of PFRun, which is not necessary present (that was the primary reason mlflow was initially used). In this code I wanted to write the universal approach for both cases, as we still will need to upload BLOB.

Reply:
However, we may move to this approach, if we will finally use pf runs always i.e. if the PF performance will be improved.

Reply:
mlflow was used only when target was not present. When target is present, we used mlflow only for logging metrics.

Reply:
Yes, I have tried to unify the logic here.